mParticle-Singular/MPKitSingular.m

```diff
@@ -346,13 +352,31 @@ - (nonnull MPKitExecStatus *) execSuccess{
 }
 
 - (void) handleOpenURLEvent:(nonnull NSURL *)url{
-    if(url){
-        SingularConfig* config = [[SingularConfig alloc] initWithApiKey:apiKey andSecret:secret];
+    if(url) {
+        SingularConfig* config = [self buildSingularConfig];
         config.openUrl = url;
-        config.singularLinksHandler = singularLinkHandler;
-        
         [Singular start:config];
     }
 }
 
+-(SingularConfig*)buildSingularConfig {
+    SingularConfig* config = [[SingularConfig alloc] initWithApiKey:apiKey andSecret:secret];
+    
+    config.singularLinksHandler = singularLinkHandler;
+    
+    config.skAdNetworkEnabled = isSKANEnabled;
+    config.manualSkanConversionManagement = isManualMode;
+    config.conversionValueUpdatedCallback = conversionValueUpdatedCallback;
+    config.waitForTrackingAuthorizationWithTimeoutInterval = waitForTrackingAuthorizationWithTimeoutInterval;
+    
+    return config;
+}
+
++(void)setSKANOptions:(BOOL)skanEnabled andMode:(BOOL)manualMode withWaitForTrackingAuthorizationWithTimeoutInterval:(NSNumber*)waitTrackingAuthorizationWithTimeoutInterval withConversionValueUpdatedHandler:(void(^_Nullable)(NSInteger))conversionValueUpdatedHandler {
```

Comment:
what if this method is called post config? then we don't care?

Reply:
This method needs to be called once before mParticle init.
If it's called afterwards we have nothing to do.